src/app/adtwizard.h

```diff
@@ -37,7 +45,24 @@ class ADTWizard : public QWizard
         Finish_Page
     };
 
-    ADTWizard(QWidget *parent = nullptr);
+    ADTWizard(QString jsonFile, QWidget *parent = nullptr);
+
+private slots:
+    void cancelButtonPressed();
+
+signals:
+    void cancelPressed(int currentPage);
+
+private:
+    std::unique_ptr<DiagnosticTool> diagnosticTool;
+
+    std::unique_ptr<IntroWizardPage> introPage;
```

Comment:
Maybe we should replace unique_ptr with QScopedPointer.

Reply:
Fixed.